src/Worker.Extensions.DurableTask/DurableHttpRequest.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Text.Json.Serialization;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Azure.Functions.Worker;
+
+/// <summary>
+/// Request used to make an HTTP call through Durable Functions.
+/// </summary>
+[JsonConverter(typeof(DurableHttpRequestConverter))]
+public class DurableHttpRequest
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DurableHttpRequest"/> class.
+    /// </summary>
+    public DurableHttpRequest(
+        HttpMethod method,
+        Uri uri,
+        IDictionary<string, StringValues>? headers = null,
+        string? content = null,
+        bool asynchronousPatternEnabled = true,
+        TimeSpan? timeout = null,
+        HttpRetryOptions? httpRetryOptions = null)
+    {
+        this.Method = method;
+        this.Uri = uri;
+        this.Headers = HttpHeadersHelper.CreateCopy(headers);
```

Comment:
I wonder what value creating a copy of the headers offers here? We never mutate them ourselves. We can probably forgo this copy and leave it to users to know not to mutate the dictionary they supply if they don't want it to change.